sunpy/coordinates/frames.py

```diff
@@ -325,6 +328,12 @@ class HeliographicCarrington(BaseHeliographic):
 
     observer = ObserverCoordinateAttribute(HeliographicStonyhurst)
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.observer == 'self' and self._is_2d:
```

Comment:
As I mentioned in Slack, the error here is because, if `self.observer` is a frame instance, the == calls `is_equivalent_frame` on the string "self", which is raising an error
```suggestion
        if not isinstance(self.observer, BaseCoordinateFrame) and coord.self.observer == 'self' and self._is_2d:
```